x/ibc/06-solomachine/types/client_state.go

```diff
@@ -0,0 +1,438 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	connectionexported "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
+	connectiontypes "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	commitmentexported "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/exported"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+)
+
+var _ clientexported.ClientState = ClientState{}
+
+// ClientState of a Solo Machine represents whether or not the client is frozen.
+type ClientState struct {
+	// Client ID
+	ID string `json:"id" yaml:"id"`
+
+	// Frozen status of the client
+	Frozen bool `json:"frozen" yaml:"frozen"`
+
+	// Current consensus state of the client
+	ConsensusState ConsensusState `json:"consensus_state" yaml:"consensus_state"`
+}
+
+// InitializeFromMsg creates a solo machine client from a MsgCreateClient
+func InitializeFromMsg(msg MsgCreateClient) (ClientState, error) {
+	return Initialize(msg.GetClientID(), msg.ConsensusState)
+}
+
+// Initialize creates an unfrozen client with the initial consensus state
+func Initialize(id string, consensusState ConsensusState) (ClientState, error) {
+	return ClientState{
+		ID:             id,
+		Frozen:         false,
+		ConsensusState: consensusState,
+	}, nil
+}
+
+// NewClientState creates a new ClientState instance.
+func NewClientState(id string, consensusState ConsensusState) ClientState {
+	return ClientState{
+		ID:             id,
+		Frozen:         false,
+		ConsensusState: consensusState,
+	}
+}
+
+// GetID returns the solo machine client state identifier.
+func (cs ClientState) GetID() string {
+	return cs.ID
+}
+
+// GetChainID returns an empty string.
+func (cs ClientState) GetChainID() string {
+	return ""
+}
```

Comment:
should we add the chain-id to the fields? @cwgoes

Reply:
I suppose, although nothing should really be dependent on it.